receiver/mongodbatlasreceiver/alerts.go

```diff
@@ -182,30 +182,50 @@ func (a *alertsReceiver) startPolling(ctx context.Context, host component.Host)
 }
 
 func (a *alertsReceiver) retrieveAndProcessAlerts(ctx context.Context) error {
-	var alerts []mongodbatlas.Alert
+	numAlertsProcessed := 0
 	for _, p := range a.projects {
 		project, err := a.client.GetProject(ctx, p.Name)
 		if err != nil {
 			a.logger.Error("error retrieving project "+p.Name+":", zap.Error(err))
 			continue
 		}
-		projectAlerts, err := a.client.GetAlerts(ctx, project.ID, a.maxAlerts)
+		a.pollAndProcess(ctx, p, project, int64(numAlertsProcessed))
+	}
+	return a.writeCheckpoint(ctx)
+}
+
+func (a *alertsReceiver) pollAndProcess(ctx context.Context, pc *ProjectConfig, project *mongodbatlas.Project, numAlertsPolled int64) {
+	pageNum := 0
+	for {
+		if a.maxAlerts <= numAlertsPolled {
+			break
+		}
+
+		projectAlerts, hasNext, err := a.client.GetAlerts(ctx, project.ID, pageNum)
 		if err != nil {
 			a.logger.Error("unable to get alerts for project", zap.Error(err))
-			continue
+			break
+		}
+
+		filteredAlerts := a.applyFilters(pc, projectAlerts)
+		now := pcommon.NewTimestampFromTime(time.Now())
+		logs, err := a.convertAlerts(now, filteredAlerts)
+		if err != nil {
+			a.logger.Error("error processing alerts", zap.Error(err))
+			break
+		}
+
+		if logs.LogRecordCount() > 0 {
+			if err = a.consumer.ConsumeLogs(ctx, logs); err != nil {
+				a.logger.Error("error consuming alerts", zap.Error(err))
+				break
+			}
+		}
+		numAlertsPolled += int64(len(projectAlerts))
```

Comment:
Do we really need to measure the number of alerts _actually_ retrieved?  We know the page size and max alerts, so we can calculate ahead of time the max number of pages, and just break as soon as `!hasNext`.

Stepping back though, I think this highlights that `max_alerts` is not giving the user enough control. It would be better to give them `page_size` and `max_pages`. They can still have effective control over the max number of alerts, but also can tune with the number of requests made to the API and size of payloads sent down the pipeline. I suspect these are the parameters that will need to be provided in the long run, so we should likely swap these in now rather than have to support both in the long run.